src/corona_hakab_model/update_matrix.py

```diff
@@ -69,8 +75,15 @@ def normalize(self):
             # saves this for the effective r0 graph
             self.total_contagious_probability = total_contagious_probability
 
+            # Random connections
+            random_total = np.dot(self.manager.num_of_random_connections.sum(0),
+                                  self.manager.random_connections_strength)
+
             # this factor should be calculated once when the matrix is full, and be left un-changed for the rest of the run.
-            self.normalize_factor = (beta * self.size) / (self.matrix.total())
+            self.normalize_factor = (beta * self.size) / (self.matrix.total() + random_total)
+
+            # Normalize random connections (only once!)
+            self.manager.random_connections_strength *= self.normalize_factor
```

Comment:
I am OK with it, but can u please explain  to me why it is needed?

Reply:
for the same reason normalizing the matrix is important. 
we want to make sure that the average amount of people a sick agent will infect will be R0. that includes both the people he regularly meets, and those he randomly meets.